client/my-sites/email/email-pricing-notice/index.tsx

```diff
@@ -0,0 +1,174 @@
+import { translate as originalTranslate, useTranslate } from 'i18n-calypso';
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+import Notice from 'calypso/components/notice';
+import type { EmailCost, ResponseDomain } from 'calypso/lib/domains/types';
+import type { ProductListItem } from 'calypso/state/products-list/selectors/get-products-list';
+import type { TranslateResult } from 'i18n-calypso';
+import type { ReactElement } from 'react';
+
+const doesAdditionalPriceMatchStandardPrice = (
+	domain: ResponseDomain,
+	mailProduct: ProductListItem,
+	purchaseCost: EmailCost | null
+): boolean => {
+	if ( ! domain ) {
+		return true;
+	}
```

Comment:
Either the code or the argument definition needs to be updated, as `! domain` can't be true if `domain` is required to be a non-nullable `ResponseDomain`.

Reply:
Good point!

Reply:
Changing the accepted types addresses the issue. :+1: